src/Microsoft.IdentityModel.Tokens/Json/JsonSerializerPrimitives.cs

```diff
@@ -695,6 +706,109 @@ internal static string ReadStringOrNumberAsString(ref Utf8JsonReader reader, str
             return retVal;
         }
 
+        internal const string TryToCreateDateTimeClaimsSwitch = "Switch.Microsoft.IdentityModel.TryAllStringClaimsAsDateTime";
+
+        public static bool TryAllStringClaimsAsDateTime()
+        {
+            return (AppContext.TryGetSwitch(TryToCreateDateTimeClaimsSwitch, out bool tryAsDateTime) && tryAsDateTime);
+        }
+
+        /// <summary>
+        /// This is a non-exhaustive list of claim types that are not expected to be DateTime values
+        /// sourced from expected Entra V1 and V2 claims, OpenID Connect claims, and a selection of
+        /// restricted claim names.
+        /// </summary>
+        private static HashSet<string> KnownNonDateTimesClaimTypes = new(StringComparer.Ordinal)
```

Comment:
We may want to think about header claims.
[JWT:](https://datatracker.ietf.org/doc/html/rfc7519#section-5) , cty, typ, [JWS](https://datatracker.ietf.org/doc/html/rfc7515#section-4.1) crit, alg, jku, jwk, kid, x5u, x5c, x5t, x5t#256, typ  [JWE](https://datatracker.ietf.org/doc/html/rfc7516#section-4.1) enc, zip

Question is do we lump payload and header in the same switch?
I think it's probably OK to have one list, we can always expose this later and give users complete control.

Reply:
added